lib/src/iterable_extensions.dart

```diff
@@ -643,6 +711,74 @@ extension IterableIntegerExtension on Iterable<int> {
 ///
 /// Specialized version of some extensions of [IterableNumberExtension].
 extension IterableDoubleExtension on Iterable<double> {
+  /// A minimal element of the iterable, or `null` it the iterable is empty.
```

Comment:
Should we handle NaNspecially? I think we should. If any element is NaN, we should return NaN, both for `min` and `max`.

We can insert
```dart
   if (value.isNaN) return value;
    ... 
     if (newValue.isNaN) return newValue;
```
after reading values from the iterator.

(I wouldn't use the behavior of `double.compareTo`, that's intended to be a complete ordering for *sorting*, and therefore puts it at one of the ends - can't remember which - but here I think returning NaN is the always right thing to do if any value is NaN).

Reply:
Done

Reply:
Added tests for this case as well.

Reply:
So, I was about to add tests for `Iterable<num>` but since it uses `compareTo` it doesn't return NaN for `.minOrNull`. If I add an extension on Iterable<num> it will conflict with the IterableComparable definition.

Reply:
An iterable on `num` should take precedence over the one on `Iterable<T = num>`.

Since `num` is a proper subtype of `Iterable<num>`, it's a more specific type and the extension conflict resolution should prefer `on Iterable<num>` to `on Iterable<Comparable<num>>`.

It's technically a breaking change to change the behavior of `Iterable<num>`, but we are already doing that for `Iterable<double>` (which would previously also match `Iterable<Comparable<num>>` and return a `num`).

I'd add the `num` version too, and handle `NaN` there as well.